### ISSUE TITLE
Counter32 and Counter64 Values Issue

### ISSUE THREAD

**Issue by ichekaldin** [swisscom/ruby-netsnmp#23]
Here is my setup.

Ruby:
```
$ ruby -v
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
```

Ubuntu:
```
$ cat /etc/issue
Ubuntu 16.04.4 LTS \n \l
```

ruby-netsnmp Gem:
```
$ gem query netsnmp

*** LOCAL GEMS ***

netsnmp (0.1.4)
```

From what I can tell, the values of Counter32 OIDs are reported incorrectly. Compare the following.

Here is an example of `ifInOctets` table returned by snmpwalk:
```
$ snmpwalk -v3 -l authNoPriv -u myuser -a sha -A "mypassword" myhost.example.com ifInOctets
IF-MIB::ifInOctets.1 = Counter32: 92484354
IF-MIB::ifInOctets.2 = Counter32: 3332448588
IF-MIB::ifInOctets.3 = Counter32: 0
IF-MIB::ifInOctets.15 = Counter32: 2569477446
IF-MIB::ifInOctets.17 = Counter32: 13813612
IF-MIB::ifInOctets.18 = Counter32: 10101063
IF-MIB::ifInOctets.19 = Counter32: 10705215
IF-MIB::ifInOctets.20 = Counter32: 61569568
IF-MIB::ifInOctets.21 = Counter32: 11202500
```

And here is the same table values returned by Sensu plugin using ruby-netsnmp gem:
```
$ ruby metrics-snmp-if.rb -h myhost.example.com -u myuser -p mypassword -l auth_no_priv | grep ifInOctets
snmp.interfaces.lo.ifInOctets 1411 1530644867
snmp.interfaces.eth0.ifInOctets 198 1530644867
snmp.interfaces.vti90.ifInOctets 153 1530644867
snmp.interfaces.vti91.ifInOctets 210 1530644867
snmp.interfaces.vti0.ifInOctets 154 1530644867
snmp.interfaces.vti1.ifInOctets 163 1530644867
snmp.interfaces.vti2.ifInOctets 939 1530644867
snmp.interfaces.vti3.ifInOctets 170 1530644867
```

I believe the simple fix is to change line 106 in lib/netsnmp/varbind.rb from:
```
        asn.value.unpack("n*")[0] || 0
```
to:
```
        asn.value.unpack("N*")[0] || 0
```
(in other words, replacing `n*` with `N*`)

Additionally, the gem currently returns nil when retrieving Counter64 OIDs. I believe the fix for that is to add the following lines 113-114 to lib/netsnmp/varbind.rb:
```
      when 6 # ASN counter 64
        asn.value.unpack("N*")[0] || 0
```

Example of my output after the fix is applied:
```
$ ruby metrics-snmp-if.rb -h myhost.example.com -u myuser -p mypassword -l auth_no_priv | grep ifInOctets
snmp.interfaces.lo.ifInOctets 92484354 1530645343
snmp.interfaces.eth0.ifInOctets 13021274 1530645343
snmp.interfaces.vti90.ifInOctets 10037049 1530645343
snmp.interfaces.vti91.ifInOctets 13820746 1530645343
snmp.interfaces.vti0.ifInOctets 10105389 1530645343
snmp.interfaces.vti1.ifInOctets 10709437 1530645343
snmp.interfaces.vti2.ifInOctets 61576026 1530645343
snmp.interfaces.vti3.ifInOctets 11206618 1530645343

$ ruby metrics-snmp-if.rb -h myhost.example.com -u myuser -p mypassword -l auth_no_priv | grep ifHCInOctets
snmp.interfaces.lo.ifHCInOctets 92484354 1530645383
snmp.interfaces.eth0.ifHCInOctets 29798688 1530645383
snmp.interfaces.vti90.ifHCInOctets 26814267 1530645383
snmp.interfaces.vti91.ifHCInOctets 13821238 1530645383
snmp.interfaces.vti0.ifHCInOctets 10105673 1530645383
snmp.interfaces.vti1.ifHCInOctets 10709721 1530645383
snmp.interfaces.vti2.ifHCInOctets 61576518 1530645383
snmp.interfaces.vti3.ifHCInOctets 11207006 1530645383
```

**Comment by ichekaldin** [comment]
FWIW, I think this is related to #21

**Comment by HoneyryderChuck** [comment]
Thx for the report. Ugh, the counters... I assume you tried the master branch already.

**Comment by HoneyryderChuck** [comment]
ouch, guilty. I might have forgotten to issue a release. Will do ti the end of the day. sorry about that.

**Comment by HoneyryderChuck** [comment]
just released 0.1.5 . Thx for the nudge again!
